statsmodels/tsa/holtwinters.py

```diff
@@ -0,0 +1,472 @@
+                         "resid", "k"])
+
+def holt_winters(data, alpha=None, beta=None, gamma=None, m=None, h=1, trend=None, 
+                 damped=False, seasonal=None, phi=None, optimised=True,
+                 boxcoxed=False, remove_bias=False, try_harder=False):  
+    """
+    Holt Winter's Exponential Smoothing
+
+    Parameters
+    ----------
+    data : array-like
+        Time series
+    alpha : float, optional
+        The alpha value of the simple exponnetial smoothing, if the value is
+        set then this value will be used as the value.
+    beta :  float, optional
+        The beta value of the holts trend method, if the value is
+        set then this value will be used as the value.
+    gamma : float, optional
+        The gamma value of the holt winters seasonal method, if the value is
+        set then this value will be used as the value.
+    m : int
+        The number of seasons to consider for the holt winters.
+    h : int
+        The number of time steps to forecast ahead.
+    trend : str {"add", "mul", None}
+        Type of trend component.
+    damped : bool
+        Should the trend component be damped.
+    seasonal : str {"add", "mul", None}
+        Type of seasonal component.
+    phi : float
+        The phi value of the damped method, if the value is
+        set then this value will be used as the value.
+    optimised : bool
+        Should the values that have not beem set above be optimised automatically?
+    boxcoxed : bool
+        Should the boxcox tranform be applied to the data first?
+    remove_bias : bool
+        Should the bias be removed from the fcast and fitted values before being
+        returned?
+    try_harder : bool
+        Should the opptimser try harder to find optimal values?\
+
+    Returns
+    -------
+    results : namedtuple
+        A namedtuple with fcast, fitted and other attributes
+
+    Notes
+    -----
+    This is a full implementation of the holt winters exponential smoothing as 
+    per [1]. This includes all the unsatble methods as well as the stable methods.
+    The implmentaion of the library follows as per the R library as much as possible
+    whilest still being pythonic.
+    
+    References
+    ----------
+    [1] Hyndman, Rob J., and George Athanasopoulos. Forecasting: principles and practice. OTexts, 2014.
+    """
+    opt=None
+    trending = trend in ['mul','add']
+    seasoning = seasonal in ['mul','add']    
+    if damped and not trending:
+        raise NotImplementedError('Can only dampen the trend component')
+    phi = phi if damped else 1.0
+    N = len(data)
+    #if not isinstance(data, pd.Series):
+        #raise NotImplementedError('Only Pandas Series Supported')
+    if boxcoxed == True:
+        y, lam6da = boxcox(data)
+    elif boxcoxed == 'log':
+        lam6da = 0.0
+        y = boxcox(data, 0.0)
+    else:
+        if isinstance(data, pd.Series):
+            y = data.values.flatten()
```

Comment:
Feels wrong to blindly flatten a (M,N) array.  Probably need a guard here -- `squeeze` and then check ndim==1.